src/core/lang/thread/AsyncRunner.hpp

```diff
@@ -59,7 +58,8 @@ template <typename Result, typename... Args> class AsyncRunner :
   auto operator=(AsyncRunner const&) noexcept = delete;
   auto operator=(AsyncRunner&&) noexcept = delete;
 
-  template <typename F> explicit(false) AsyncRunner(F&& function) noexcept(false) : _fn(std::forward<F>(function)) {}
+  template <meta::concepts::DifferentFrom<AsyncRunner> F> explicit(false) AsyncRunner(F&& function) noexcept(false) :
```

Comment:
@LoghinVladDev why the need for F to be different from AsyncRunner?

Reply:
Avoid hiding the deleted copy/move ctor.
Even if `AsyncRunner(F&& func)` [with `F&&` = `AsyncRunner` "`const&`"/"`&&`"] does lead to a sfinae error from `_fn(std::forward<F>(function))`, since `cds::Function` requires a func sig / callable.

Since `AsyncRunner` does not satisfy `Callable` (has `operator()`), error would not be immediately clear.
With concepts, it's clearer.

Moreover, what if, in the future, `AsyncRunner` would have a `operator()`? We would have a non-easily detectable error, probably leading to UB or other funny stuff